serial/tools/list_ports_linux.py

```diff
@@ -88,22 +88,19 @@ def read_line(self, *args):
             return None
 
 
-def comports(include_links=False):
+def comports(include_links=False, hide_subsystems=["platform"]):
     devices = set()
-    devices.update(glob.glob('/dev/ttyS*'))     # built-in serial ports
-    devices.update(glob.glob('/dev/ttyUSB*'))   # usb-serial with own driver
-    devices.update(glob.glob('/dev/ttyXRUSB*')) # xr-usb-serial port exar (DELL Edge 3001)
-    devices.update(glob.glob('/dev/ttyACM*'))   # usb-serial with CDC-ACM profile
-    devices.update(glob.glob('/dev/ttyAMA*'))   # ARM internal port (raspi)
-    devices.update(glob.glob('/dev/rfcomm*'))   # BT serial devices
-    devices.update(glob.glob('/dev/ttyAP*'))    # Advantech multi-port serial controllers
-    devices.update(glob.glob('/dev/ttyGS*'))    # https://www.kernel.org/doc/Documentation/usb/gadget_serial.txt
+    drivers = open('/proc/tty/drivers').readlines()
+    for driver in drivers:
+        items = driver.strip().split()
+        if items[4] == 'serial':
```

Comment:
it would be better to ensure that this method works and fallback to the original version, this [4] access scares me a bit.

Reply:
I've opened a new MR with this, and a few other updates

Reply:
We can certainly check the length first, though every line should hold 5 values.
But other than that I don't see any other way of accessing that column in /proc/tty/drivers
And I also don't see the difference between column 4 and column 1 when it comes to trusting its value.